### PR TITLE
STELLAR-2726 Groundstation api doppler block

### PR DESCRIPTION
**Pull request by reiinakano**
Will add more tests  + waiting on new release of stellarstation-api v0.3.0

### CODE REVIEW

File: gr-starcoder/python/groundstation_api_doppler.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+from google.auth import jwt as google_auth_jwt
+from google.auth.transport import grpc as google_auth_transport_grpc
+from google.protobuf.timestamp_pb2 import Timestamp
+from stellarstation.api.v1.groundstation import groundstation_pb2
+from stellarstation.api.v1.groundstation import groundstation_pb2_grpc
+
+class groundstation_api_doppler(gr.sync_block):
+    """
+    This block communicates with the Ground Station API server, finds the specified plan ID, and sends out the
+    appropriate Doppler shifts at the specified interpolated interval.
```
Comment (kenichi-fukushima) [inline gr-starcoder/python/groundstation_api_doppler.py]: From this description I can't tell how to consume output of this block. Is a Doppler shift an amount in the first place?

File: gr-starcoder/python/groundstation_api_doppler.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    api_key (str): Path to API key file for authentication. If empty string, use insecure channel.
+    api_url (str): URL to gRPC API.
+    root_cert_path (str): Root certificate path for TLS. If empty string, use insecure channel.
+    groundstation_id (str): Groundstation ID
+    plan_id (str): Plan ID
+    corrections_per_second (int): Number of times the Doppler shift is sent out per second.
```
Comment (kenichi-fukushima) [inline gr-starcoder/python/groundstation_api_doppler.py]: The description above mentions "interval". This parameter should match that.

File: gr-starcoder/python/groundstation_api_doppler.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    verbose (bool): Controls verbosity
```
Comment (kenichi-fukushima) [inline gr-starcoder/python/groundstation_api_doppler.py]: `Enables verbose logging` or something.

**Comment by kenichi-fukushima** [comment]
stellarstation-api 0.4.0 was published.
https://pypi.org/project/stellarstation/

**Comment by reiinakano** [comment]
No longer WIP. PTAL

File: gr-starcoder/python/groundstation_api_doppler.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+from stellarstation.api.v1.groundstation import groundstation_pb2_grpc
+
+SPEED_OF_LIGHT = 299792458.0
+
+class groundstation_api_doppler(gr.sync_block):
+    """
+    This block communicates with the Ground Station API server, finds the specified plan ID, and sends out the
+    Doppler shifted frequencies of the satellite at the specified number of corrections per second.
+
+    api_key (str): Path to API key file for authentication. If empty string, use insecure channel.
```
Comment (kenichi-fukushima) [inline gr-starcoder/python/groundstation_api_doppler.py]: Please update this list with new parameters.

File: gr-starcoder/python/groundstation_api_doppler.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        listPlansResponse = self.api_client.ListPlans(groundstation_pb2.ListPlansRequest(
+            ground_station_id=self.groundstation_id,
+            aos_after=Timestamp(seconds=int(time.time()) - 120, nanos=0),
+            aos_before=Timestamp(seconds=int(time.time() + 3600), nanos=0),
+        ))
+        self.log.debug("Fetched {} plans".format(len(listPlansResponse.plan)))
+
+        plan = [x for x in listPlansResponse.plan if x.plan_id == self.plan_id]
+        if not plan:
+            self.log.debug("No plans matching plan ID {} found.".format(self.plan_id))
```
Comment (kenichi-fukushima) [inline gr-starcoder/python/groundstation_api_doppler.py]: This is a fatal error. Log it with an appropriate level and terminate the program.

File: gr-starcoder/python/groundstation_api_doppler.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                channel = google_auth_transport_grpc.secure_authorized_channel(
+                    jwt_creds, None, self.api_url, channel_credential)
+        else:
+            channel = self.test_channel
+        return groundstation_pb2_grpc.GroundStationServiceStub(channel)
+
+    def work(self, input_items, output_items):
+        return len(output_items[0])
+
+    def handle_stream(self):
```
Comment (kenichi-fukushima) [inline gr-starcoder/python/groundstation_api_doppler.py]: I don't think `handle_stream` describes what this method does.

**Comment by reiinakano** [comment]
Merging because CircleCI is very flaky compared to Quay's build environment

**Pull request merged**